### ISSUE TITLE
Recognizing touch outside bounds of inputToolbar

### ISSUE THREAD

**Issue by brbgyn** [jessesquires/JSQMessagesViewController#1155]
Hi, i did some customization and added a view that is beyond the bounds of inputToolbar.

I did all sort of research and it is a known behavior, and is expected, but i wish i could recognize this touches so the user can scroll the tableView that is in this UIView and select a cell.

Here is a screenshot of what i'm talking about:

![ios simulator screen shot 12 08 2015 17 32 59](https://cloud.githubusercontent.com/assets/6548984/9235943/47cdb720-4118-11e5-9f5b-5d7dd365e01f.png)

In the screenshot example, in the tableview above the inputToolbar, i can touch/drag when i touch the "brunosilva" item, but not in the other items that are outside inputToolbar's bounds (bruscagin89, Bruno Borges, Bruno Vieira). The inputToolbar's frame is dark in the screenshot.

It's a subView of inputToolbar.

I've tried tons of ways to implement ```hitTest:withEvent:```. Tried it in JSQMessagesToolbarContentView, in JSQMessagesCollectionView (since it reaches the collectionView), in the JSQMessagesViewController , and in my view controller itself. No lucky.

**Comment by brbgyn** [comment]
*I added it as a subView of inputToolbar so that it can slide up and down according to keyboard animations...

**Comment by harlanhaskins** [comment]
You might want to add it as a subview of the View Controller's view and constrain it to the bottom of the input toolbar.

**Comment by brbgyn** [comment]
Hi, @harlanhaskins 

Tried what you said.

Using the following in JSQMessagesViewController:

```
NSLayoutConstraint *myConstraint = [NSLayoutConstraint constraintWithItem:floatingController.view
                                                                    attribute:NSLayoutAttributeBottom
                                                                    relatedBy:NSLayoutRelationEqual
                                                                       toItem:self.inputToolbar
                                                                    attribute:NSLayoutAttributeBottom
                                                                   multiplier:1
                                                                     constant:0];
[self.view addConstraint:myConstraint];
```

The floatingController view remains centralized, and i get a bunch of warnings logs:

```
Unable to simultaneously satisfy constraints.
	Probably at least one of the constraints in the following list is one you don't want. Try this: (1) look at each constraint and try to figure out which you don't expect; (2) find the code that added the unwanted constraint or constraints and fix it. (Note: If you're seeing NSAutoresizingMaskLayoutConstraints that you don't understand, refer to the documentation for the UIView property translatesAutoresizingMaskIntoConstraints) 
(
    "<NSLayoutConstraint:0x78e318c0 V:[JSQMessagesInputToolbar:0x788818e0]-(274)-|   (Names: '|':UIView:0x78e43db0 )>",
    "<NSAutoresizingMaskLayoutConstraint:0x78ec6450 h=-&- v=-&- UIView:0x78743900.midY == UIView:0x78e43db0.midY + 30>",
    "<NSAutoresizingMaskLayoutConstraint:0x78ec6480 h=-&- v=-&- UIView:0x78743900.height == UIView:0x78e43db0.height>",
    "<NSLayoutConstraint:0x786e3d10 UIView:0x78743900.bottom == JSQMessagesInputToolbar:0x788818e0.bottom>"
)

Will attempt to recover by breaking constraint 
<NSLayoutConstraint:0x786e3d10 UIView:0x78743900.bottom == JSQMessagesInputToolbar:0x788818e0.bottom>

Make a symbolic breakpoint at UIViewAlertForUnsatisfiableConstraints to catch this in the debugger.
```

I've never studied auto layouts (including programmatically) so i do not know if what i did was right.

Was it ?

**Comment by harlanhaskins** [comment]
You need to call [floatingController.view setTranslatesAutoresizingMaskIntoConstraints:NO] first.

Also you really should look into auto layout.

**Comment by brbgyn** [comment]
The auto-layout worked using setTranslatesAutoresizingMaskIntoConstraints.
I had to make another 3 NSLayoutConstraint because i was getting AMBIGUOUS LAYOUT in lldb.

BUT a strange thing is occurring.

I'm adding floatingController.view in self.view, but, when i set the Constraints, floatingController.view goes to self.inputToolbar, and the touches cannot be recognized again...

Is it possible that auto layout is changing the subview tree ? I added it to a view and auto-layout changed it to a subview of this view.

It's like i never added constraints.
